moor_generator/lib/src/analyzer/sql_queries/query_analyzer.dart

```diff
@@ -28,10 +30,24 @@ abstract class BaseAnalyzer {
     if (_engine == null) {
       _engine = step.task.session.spawnEngine();
       tables.map(mapper.extractStructure).forEach(_engine.registerTable);
+      resolveViews();
+      views.map(mapper.extractView).forEach(_engine.registerView);
     }
     return _engine;
   }
 
+  /// Parses the view and adds columns to its resolved columns.
+  @protected
+  void resolveViews() {
+    for (final view in views) {
+      final ctx = _engine.analyzeNode(
+          view.declaration.node, view.declaration.createSql);
```

Comment:
This is kind of a bad API from me, but the second parameter to `analyzeNode` needs to be the _full source_ containing the view declaration (so the enclosing moor file in this case).

Reply:
My bad, I can change the the second parameter to `view.file` and add a `ParsedMoorFile file;` to `MoorView`.